http/preview.go

```diff
@@ -152,5 +152,6 @@ func createPreview(imgSvc ImgService, fileCache FileCache,
 }
 
 func previewCacheKey(fPath string, fTime int64, previewSize PreviewSize) string {
+        fPath = filepath.Base(fPath)
```

Comment:
Filenames are not unique and can result in multiple files having the same cache key, the full path should be used instead.

Reply:
> 
But we also use fTime. It includes "Y,m,d,h,m,s",  as I tested, fTime alone is already enough to tell defferent images. 
And filenames is better than fullpath because when we move a large folder to another place, or make a copy of large folder, it will not generate the cache files again.

Reply:
I agree with @ramiresviana. https://pkg.go.dev/github.com/spf13/afero#BasePathFs.RealPath should be used instead.

Reply:
Fixed again. I just use this one: d.user.Scope + fPath = fullpath

Reply:
What is the behavior of that implementation when the scope is empty?

Reply:
> on when the scope is empty?

I just fixed it. Using this:         fPath = filepath.Clean("/"+ d.user.Scope + fPath)
Filebrowser's root path can be . or / or  /// or ./// or /.// or empty, 
In general, if the path starts with .***, filepath.Clean will give us a path like a/b/c... ,
if the path starts with /***, filepath.Clean will give us a path like /a/b/c...
So, we just add a "/" in front of the path, then we will get the path with one format, no matter how the scope starts with.

Reply:
You're making use of relative file paths. We need the real file path, relative to the OS filesystem root, to ensure that users will hit the correct cache entries and there will be no duplicates. Afero `RealPath()` should be able to solve it and ensure that will work on different platforms.

Reply:
Yes. ramiresviana, you mean it will be a bug when filebrowser runs in docker or something like that? 
I just don't know how to use this RealPath function. name string is the filename? It's not unique. (I know little about goLang.)
func (b *BasePathFs) RealPath(name string) (path string, err error) {
	if err := validateBasePathName(name); err != nil {
		return name, err
	}

	bpath := filepath.Clean(b.path)
	path = filepath.Clean(filepath.Join(bpath, name))
	if !strings.HasPrefix(path, bpath) {
		return name, os.ErrNotExist
	}

	return path, nil
}

func validateBasePathName(name string) error {
	if runtime.GOOS != "windows" {
		// Not much to do here;
		// the virtual file paths all look absolute on *nix.
		return nil
	}

	// On Windows a common mistake would be to provide an absolute OS path
	// We could strip out the base part, but that would not be very portable.
	if filepath.IsAbs(name) {
		return os.ErrNotExist
	}

	return nil
}

Reply:
I think we can have something like:
```go
func previewCacheKey(file *files.FileInfo, previewSize PreviewSize) (string, error) {	
	if fPath, err = file.Fs.(*afero.BasePathFs).RealPath(file.Path); err != nil {
		return nil, err
	}
	return fmt.Sprintf("%x%x%x", fPath, file.ModTime.Unix(), previewSize), nil
}
```
The usage should be:
```go
if cacheKey, err := previewCacheKey(file, previewSize); err != nil {
	return nil, err
}
```

Reply:
Fixed.
Now we just get the path like "D:\a\b\1.jpg", it's realpath.

When I use "return nil, err", it fails to compile, "nil is not an int or string". So I make some changes. I am not sure if it's correct. But it works.